### DISCUSSION TITLE
Bug : The transform tool is also squeezing stroke widths.

### DISCUSSION THREAD

**Discussion by ArtistBirdy** [keiwando/toonsquid#90] [Q&A] [answered]
- When I try to transform a vector shape using FREE-FORM mode, its stroke becomes non-uniform. 

https://user-images.githubusercontent.com/118920457/205468389-407db4b9-3597-4b03-9bac-0a0414ddc657.MOV

- It does not in other vector-based apps:

https://user-images.githubusercontent.com/118920457/205468426-e7ef2a54-37c8-456f-b6ec-4033e00b885f.MOV

**Comment by keiwando** [comment, marked as answer]
Hi!
The transform tool has two modes for pixel layers and vector / path layers. It either edits the transform properties (position, rotation and scale) or the contents of the layer. In your case, the "Edit path" option at the top is disabled, so the transform tool is editing the scale property, which scales the entire layer, so the stretching is expected.
If you enable the "Edit path option", it will only edit the control points of the shape inside of the layer, which keeps properties such as the stroke width or the corner radius etc unmodified (the behaviour of your second video).
